### PR TITLE
Add playlist random selection from category

### PR DESCRIPTION
**Pull request by maxenced**
This PR use the /category/ Spotify endpoint to pick up and play a random playlist of a given category.
I did not do it yet, but it should be fairly easy to create an entity which expose all the available categories.

3 new parameters are added, respectively:
* category : the category name
* country : 2 letter code of country to filter category
* limit : maximum number of playlist to return from api. Component will pick one randomly among them

Debug log level will show 2 lines:

```
home-assistant   | 2021-12-10 17:36:25 DEBUG (SyncWorker_4) [custom_components.spotcast.helpers] Get random playlist from 50 playlists of category chill, in country FR.
home-assistant   | 2021-12-10 17:36:25 DEBUG (SyncWorker_4) [custom_components.spotcast.helpers] Choosed playlist Peaceful Guitar (spotify:playlist:37i9dQZF1DX0jgyAiPl8Af) from category chill.
```

Maxence

### CODE REVIEW

File: custom_components/spotcast/helpers.py [hunk 1]
```diff
@@ -114,3 +115,13 @@ def get_search_results(search, spotify_client):
     _LOGGER.debug("Best match for %s is %s", search, bestMatch['name'])
 
     return bestMatch['uri']
+
+def get_random_playlist_from_category(spotify_client, category, country=None, limit=20):
+    _LOGGER.debug(f"Get random playlist among {limit} playlists from category {category} in country {country}")
+    playlists = spotify_client.category_playlists(category_id=category, country=country, limit=limit)["playlists"]["items"]
+    chosen = random.choice(playlists)
+    _LOGGER.debug(f"Choosed playlist {chosen['name']} ({chosen['uri']}) from category {category}.")
```
Comment (hmmbob) [inline custom_components/spotcast/helpers.py:123]:
```suggestion
    _LOGGER.debug(f"Chose playlist {chosen['name']} ({chosen['uri']}) from category {category}.")
```

**Comment by Darkfull-Dante** [comment]
Hi @maxenced,

I'm looking into your PR. I'll be able to finalise my review in the following days (it's currently the end of semester where I live, so I didn't have a lot of time). The feature is really interesting, but I think that adding the country code should be put to better use. I'll look into the Spotify documentation to check for any other api call we make that could benefit from location based filtering.

Also, I'm wondering if there is a benefit in having a per call setting for the country code. I'm thinking of multiple different user experiences that we could go for:

- per call request (keep as is)
- permanent param for spotcast set by the user on install (just like the sp_dc and sp_key)
- same as second + a boolean setting telling spotcast to use or not the country code (ignored for function that require it)

At the moment, especially if we start using the country code on other calls, I think the third option would be the most interesting of the bunch. Unless there is a use for the user to be able to change country per calls.

**Comment by maxenced** [comment]
Hi @Darkfull-Dante . I agree for the country code. However, I'm not sure having it global (only) is the most interesting. People may want to set it to different value to get different playlists. Maybe being able to set a default value globally and overwrite it on a per-call basis ?

**Comment by Darkfull-Dante** [comment]
Hi @maxenced, that's a really good idea, we could make the country code an optional global setting and also provide a country setting in the service call (which would be optional also). We could then use the following truth table

| Global | Service_call | Result |
|:--------:|:-----------------:|:---------|
|           |                       | No localisation, fails with error log when parameter is mandatory |
| X        |                       | Use global localisation for all calls that can use them                   |
|           | X                    | Use provided localisation for calls that can use them                    |
| X       | X                    | Ignores the global setting in favor of the service call parameter    |

I'm going to create a branch for the project. We'll be able to both work on the code at the same time. But don't worry, it's your idea, and your project, I'll let you lead for the decision for the implementation.

**Comment by Darkfull-Dante** [comment]
@maxenced, I moved the PR to the random_category branch and started the validation process, if it passes, I'll merge your fork in this branch. For any changes, simply push the PR to that branch until we have a complete solution

**Pull request merged**